src/test/java/org/cactoos/scalar/AvgOfTest.java

```diff
@@ -51,213 +51,313 @@ public void withEmptyCollection() {
+                1, 2, 3, 4
             ).floatValue(),
             Matchers.equalTo(2.5f)
         );
     }
 
     @Test
-    public void withLongCollection() {
+    public void withLongCollectionIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1L, 2L, 3L, 4L).toArray(new Long[4])
+                1L, 2L, 3L, 4L
             ).intValue(),
             Matchers.equalTo(2)
         );
+    }
+
+    @Test
+    public void withLongCollectionLongValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1L, 2L, 3L, 4L).toArray(new Long[4])
+                1L, 2L, 3L, 4L
             ).longValue(),
             Matchers.equalTo(2L)
         );
+    }
+
+    @Test
+    public void withLongCollectionMaxValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1L, 2L, 3L, 4L).toArray(new Long[4])
+                Long.MAX_VALUE, Long.MAX_VALUE
+            ).longValue(),
+            Matchers.equalTo(Long.MAX_VALUE)
+        );
+    }
+
+    @Test
+    public void withLongCollectionDoubleValue() {
+        MatcherAssert.assertThat(
+            new AvgOf(
+                1L, 2L, 3L, 4L
             ).doubleValue(),
             Matchers.equalTo(2.5d)
         );
+    }
+
+    @Test
+    public void withLongCollectionFloatValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1L, 2L, 3L, 4L).toArray(new Long[4])
+                1L, 2L, 3L, 4L
             ).floatValue(),
             Matchers.equalTo(2.5f)
         );
     }
 
     @Test
-    public void withDoubleCollection() {
+    public void withDoubleCollectionIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0d, 2.0d, 3.0d, 4.0d).toArray(new Double[4])
+                1.0d, 2.0d, 3.0d, 4.0d
             ).intValue(),
             Matchers.equalTo(2)
         );
+    }
+
+    @Test
+    public void withDoubleCollectionLongValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0d, 2.0d, 3.0d, 4.0d).toArray(new Double[4])
+                1.0d, 2.0d, 3.0d, 4.0d
             ).longValue(),
             Matchers.equalTo(2L)
         );
+    }
+
+    @Test
+    public void withDoubleCollectionDoubleValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0d, 2.0d, 3.0d, 4.0d).toArray(new Double[4])
+                1.0d, 2.0d, 3.0d, 4.0d
             ).doubleValue(),
             Matchers.equalTo(2.5d)
         );
+    }
+
+    @Test
+    public void withDoubleCollectionMaxValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0d, 2.0d, 3.0d, 4.0d).toArray(new Double[4])
-            ).floatValue(),
-            Matchers.equalTo(2.5f)
+                Double.MAX_VALUE, Double.MAX_VALUE
```

Comment:
@allanws Add tests using positive and negative numbers. And positive or negative and with NaN numbers.

Reply:
They are already there, have a look again.